### PR TITLE
enh(powershell) major overhaul, huge improvements 

### PR DESCRIPTION
**Pull request by yyyc514**
Based on #1386 (on current master)

Copying notes over:

---

As of now, the PowerShell has come a long way from being just a simple scripting language.
It currently has lots of features, like Classes, Enums, Strong Typing etc.

This commit is a major overhaul of the previous one.

**Rich highlighting enhancements were introduced to the following language structures:**

- [x] Approved Verbs - [See.](https://msdn.microsoft.com/en-us/library/ms714428(v=vs.85).aspx)
  No more need to list a long sheet of built-in commands,
  If the cmdlet is valid it will be highlighted, and if not,
  well - it will be marked as such.
- [x] Assemblies import.
- [x] Types.
- [x] Static classes reference.
- [x] <del>Methods invocation.</del>
- [x] Enums.
- [x] Classes.
- [x] <del>Constructors.</del>
- [x] Methods.
- [x] "$this" service keyword.
- [ ] Escaped $ variables.
- [x] Objects creation.
- [x] Comparison operators. Also,lots of new ones were added!
- [x] Functions with arguments.
- [ ] Named function parameters.
- [ ] Function parameters validators.
- [x] Hash tables.
- A bunch of small tweaks here and there.

### CODE REVIEW

**Comment by yyyc514** [comment]
Reviving this since it's SO much better than what we had before.  Just needs a few examples in `markup` and to figure out why tests are broken.

File: test/markup/index.js [hunk 1]
```diff
@@ -32,6 +32,10 @@ function testLanguage(language) {
 }
 
 describe('hljs.highlight()', async () => {
+  // TODO: why?
+  // ./node_modules/.bin/mocha test/markup
+  it("needs this or it can't be run stand-alone", function() {} );
+
```
[the remark this answers is not in the document]
Comment (yyyc514) [reply test/markup/index.js]: You tell me.  Without it I can't run the tests with:

```
./node_modules/.bin/mocha -t 1000  --globals document test/markup
```

It says "0 tests" for some reason.

File: tools/developer.html [hunk 2]
```diff
@@ -4,7 +4,7 @@
   <meta charset="utf-8">
   <title>highlight.js developer</title>
 
-  <link rel="stylesheet" href="../src/styles/default.css">
+  <link rel="stylesheet" href="../src/styles/vs2015.css">
```
[the remark this answers is not in the document]
Comment (yyyc514) [reply tools/developer.html]: Cause it's easier to see.  I can revert this, wasn't intentional.

File: test/markup/index.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (yyyc514) [reply test/markup/index.js]: I added the TODO so someone else could tell me how to fix it hopefully. :)

[the remark this answers is not in the document]
Comment (egor-rogov) [reply test/markup/index.js]: Hm, really. No idea, let it remain so.

[the remark this answers is not in the document]
Comment (yyyc514) [reply test/markup/index.js]: Thanks. :)

[the remark this answers is not in the document]
Comment (egor-rogov) [reply test/markup/index.js]: We should thank you! (:

**Comment by n8tb1t** [comment]
It took like 4 years since I published the #1386, but anyways I'm glad it's finally merged :)

**Comment by yyyc514** [comment]
Better late than never they say. 😬

**Pull request merged**